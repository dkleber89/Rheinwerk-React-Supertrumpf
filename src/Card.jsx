import * as React from 'react';
import PropTypes from 'prop-types';

import './Card.css';
import Animal from './Animal';
import DarkMode from './DarkMode';
import { useContext } from 'react';

export default function Card({ animal, uncovered, onSelectProperty, selectedProperty }) {
  const front = (
    <div className="card">
      <h1>{animal.name ? animal.name : 'Unbekannt'}</h1>
      {animal.image && <img alt="Elefant" height="200" width="200" src={`${process.env.PUBLIC_URL}/${animal.image}`} />}
      <table>
        <tbody>
          {Object.keys(Animal.properties).map(property => {
            const animalProperty = Animal.properties[property];

            return (
              <tr key={property} className={selectedProperty === property ? 'active' : ''} onClick={() => onSelectProperty(property)}>
                <td>{animalProperty.label}</td>
                <td>{`${animal[property]} ${animalProperty.unit}`}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );

  const back = <div className="card back" />;

  const darkMode = useContext(DarkMode);
  const darkModeClassName = darkMode ? 'dark' : 'light';

  return <div className={darkModeClassName}>{uncovered ? front : back}</div>;
}

Card.propTypes = {
  uncovered: PropTypes.bool.isRequired,
  animal: PropTypes.instanceOf(Animal).isRequired,
  onSelectProperty: PropTypes.func,
  selectedProperty: PropTypes.string,
};
