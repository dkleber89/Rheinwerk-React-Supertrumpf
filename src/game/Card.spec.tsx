import React from 'react';
import renderer from 'react-test-renderer';
import Animal from './Animal';
import Card from './Card';

describe('Card', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const animal = new Animal('Elefant', 'Bild', 3.3, 6000, 70, 1, 40);

      const tree = renderer.create(<Card animal={animal} uncovered={true}/>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
