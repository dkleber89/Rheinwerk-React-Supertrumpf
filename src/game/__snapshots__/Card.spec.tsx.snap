// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots should match snapshot 1`] = `
<div
  className="card light"
>
  <h1>
    Elefant
  </h1>
  <img
    alt="Elefant"
    height="200"
    src="/Bild"
    width="200"
  />
  <table
    className="sc-AxiKw hrgZiT"
  >
    <tbody>
      <tr
        className="sc-AxirZ ktjwNY size"
        onClick={[Function]}
      >
        <td
          className="sc-AxjAm iROnEI"
        >
          Größe
        </td>
        <td
          className="sc-AxjAm iROnEI"
        >
          3.3
           
          m
        </td>
      </tr>
      <tr
        className="sc-AxirZ ktjwNY weight"
        onClick={[Function]}
      >
        <td
          className="sc-AxjAm iROnEI"
        >
          Gewicht
        </td>
        <td
          className="sc-AxjAm iROnEI"
        >
          6000
           
          kg
        </td>
      </tr>
      <tr
        className="sc-AxirZ ktjwNY age"
        onClick={[Function]}
      >
        <td
          className="sc-AxjAm iROnEI"
        >
          Alter
        </td>
        <td
          className="sc-AxjAm iROnEI"
        >
          70
           
          Jahre
        </td>
      </tr>
      <tr
        className="sc-AxirZ ktjwNY offspring"
        onClick={[Function]}
      >
        <td
          className="sc-AxjAm iROnEI"
        >
          Nachkommen
        </td>
        <td
          className="sc-AxjAm iROnEI"
        >
          1
           
          
        </td>
      </tr>
      <tr
        className="sc-AxirZ ktjwNY speed"
        onClick={[Function]}
      >
        <td
          className="sc-AxjAm iROnEI"
        >
          Geschwindigkeit
        </td>
        <td
          className="sc-AxjAm iROnEI"
        >
          40
           
          km/h
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
